parser-typechecker/src/Unison/Typechecker/Context.hs

```diff
@@ -1631,15 +1636,15 @@ verifyClosedTerm t = do
   ok1 <- verifyClosed t id
   let freeTypeVars = Map.toList $ Term.freeTypeVarAnnotations t
       reportError (v, locs) = for_ locs $ \loc ->
-        recover () (failWith (UnknownSymbol loc (TypeVar.underlying v)))
```

Comment:
Good catch. I thought this `for_` was using the `Applicative` instance  for `Result v loc`, which accumulates errors (unlike the `Monad` instance), but it is an `Applicative` for `M v loc`, which currently does not. To fix, it suffices to write out the implementation of `Applicative (M v loc)` using ops of `Applicative (Result v loc)`, instead of just deferring to the `Monad` instance. Will do it later today.

Reply:
Okay, `Applicative (M v loc)` that accumulates errors is not possible, because `M` updates state (of type `Env`), which must be done sequentially.

However, the fix I implemented is even better: do work with `Result` where the full power of `M` is not necessary and use its `Applicative` to accumulate errors. This way, we can accumulate even more errors, as in

```haskell
      liftResult $  verifyDataDeclarations datas
                 *> verifyDataDeclarations (DD.toDataDecl <$> effects)
                 *> verifyClosedTerm term
```

Reply:
Also, I added a test that all unknown symbols are reported.

Reply:
Great, thanks!